noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -1,9 +1,75 @@
-use dep::protocol_types::abis::event_selector::EventSelector;
+use crate::{
+    context::{PrivateContext, PublicContext},
+    messages::logs::{event::to_encrypted_private_event_message, utils::prefix_with_tag},
+    utils::remove_constraints::remove_constraints_if,
+};
+use dep::protocol_types::{
+    abis::event_selector::EventSelector,
+    address::AztecAddress,
+    traits::{Serialize, ToField},
+};
 
-pub trait EventInterface {
-    fn get_event_type_id() -> EventSelector;
+pub struct PrivateLogContentConstraints {
+    /// The contents of the log are entirely unconstrained, and could have any values.
+    ///
+    /// Only use in scenarios where the recipient not receiving the message is an acceptable outcome (e.g. because the
+    /// sender is somehow motivated to ensure the recipient learns of it).
+    pub NO_CONSTRAINTS: u8,
+    /// The contents of the log and its encryption are constrained. The tag (and therefore whether the recipient is
+    /// actually able to find the message) is not.
+    ///
+    /// Only use in scenarios where the recipient not receiving the message is an acceptable outcome (e.g. because the
+    /// sender is somehow motivated to ensure the recipient learns of it).
+    pub CONSTRAINED_ENCRYPTION: u8,
```

Comment:
the docs for this are the same as the docs for `NO_CONSTRAINTS` since this variant doesn't actually make sense: we need to also constrain tagging for constrained encryption to be useful (but not handshaking!). I'll add a proper TODO once I'm done here.